File: sdk/go/mfa.go
package qeetid

import (
	"context"
	"net/http"
	"net/url"
)

type MfaFactor struct {
	ID        string `json:"id"`
	UserID    string `json:"user_id"`
	Type      string `json:"type"`
	Status    string `json:"status"`
	CreatedAt string `json:"created_at"`
}

type MfaAdmin struct{ http *httpClient }

func (r *MfaAdmin) List(ctx context.Context, userID string) ([]MfaFactor, error) {
	var env struct {
		Items []MfaFactor `json:"items"`
		Data  []MfaFactor `json:"data"`
	}
	if err := r.http.do(ctx, http.MethodGet, "/v1/users/"+url.PathEscape(userID)+"/mfa", nil, nil, &env, true); err != nil {
		return nil, err
	}
	if env.Items != nil {
		return env.Items, nil
	}
	return env.Data, nil
}

func (r *MfaAdmin) Reset(ctx context.Context, userID string) error {
	return r.http.do(ctx, http.MethodDelete, "/v1/users/"+url.PathEscape(userID)+"/mfa", nil, nil, nil, true)
}

func (r *MfaAdmin) Require(ctx context.Context, userID, tenantID string) error {
	body := map[string]string{"tenant_id": tenantID}
	return r.http.do(ctx, http.MethodPost, "/v1/users/"+url.PathEscape(userID)+"/mfa/require", nil, body, nil, false)
}

File: sdk/js/node/src/mfa.ts
import type { HttpClient } from "./client.js";

export interface MfaFactor {
  id: string;
  user_id: string;
  type: string;
  status: string;
  created_at: string;
}

export class MfaAdmin {
  constructor(private readonly http: HttpClient) {}

  async list(userId: string): Promise<MfaFactor[]> {
    const res = await this.http.get<{ items?: MfaFactor[]; data?: MfaFactor[] }>(
      `/v1/users/${encodeURIComponent(userId)}/mfa`,
    );
    return res.items ?? res.data ?? [];
  }

  reset(userId: string): Promise<void> {
    return this.http.delete<void>(`/v1/users/${encodeURIComponent(userId)}/mfa`);
  }

  require(userId: string, tenantId: string): Promise<void> {
    return this.http.post<void>(`/v1/users/${encodeURIComponent(userId)}/mfa/require`, {
      tenant_id: tenantId,
    });
  }
}
